org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/contentassist/ContentAssistEntry.java

```diff
@@ -0,0 +1,289 @@
+/**
+ * Copyright (c) 2015, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.ide.editor.contentassist;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.xtext.util.ReplaceRegion;
+import org.eclipse.xtext.util.TextRegion;
+import org.eclipse.xtext.xbase.lib.util.ToStringBuilder;
+
+/**
+ * @noreference
+ */
+public class ContentAssistEntry {
+
+	public static final String KIND_TEXT = "TEXT";
+
+	public static final String KIND_METHOD = "METHOD";
+
+	public static final String KIND_FUNCTION = "FUNCTION";
+
+	public static final String KIND_CONSTRUCTOR = "CONSTRUCTOR";
+
+	public static final String KIND_FIELD = "FIELD";
+
+	public static final String KIND_VARIABLE = "VARIABLE";
+
+	public static final String KIND_CLASS = "CLASS";
+
+	public static final String KIND_INTERFACE = "INTERFACE";
+
+	public static final String KIND_MODULE = "MODULE";
+
+	public static final String KIND_PROPERTY = "PROPERTY";
+
+	public static final String KIND_UNIT = "UNIT";
+
+	public static final String KIND_VALUE = "VALUE";
+
+	public static final String KIND_ENUM = "ENUM";
+
+	public static final String KIND_KEYWORD = "KEYWORD";
+
+	public static final String KIND_SNIPPET = "SNIPPET";
+
+	public static final String KIND_COLOR = "COLOR";
+
+	public static final String KIND_FILE = "FILE";
+
+	public static final String KIND_REFERENCE = "REFERENCE";
+
+	public static final String KIND_UNKNOWN = "UNKNOWN";
+
+	/**
+	 * The prefix that should be replaced with this proposal.
+	 */
+	private String prefix;
+
+	/**
+	 * The proposed text to be inserted.
+	 */
+	private String proposal;
+
+	/**
+	 * The text seen by the user in the list of proposals.
+	 */
+	private String label;
+
+	/**
+	 * Additional description to include in the list of proposals.
+	 * <p>
+	 * This property may not be supported by all editor frameworks.
+	 * </p>
+	 */
+	private String description;
+
+	/**
+	 * Documentation for the proposal proposals.
+	 * <p>
+	 * This property may not be supported by all editor frameworks.
+	 * </p>
+	 */
+	private String documentation;
+
+	/**
+	 * The absolute cursor position to apply after the proposal has been inserted. If omitted, the cursor it set to the
+	 * end of the inserted proposal.
+	 * <p>
+	 * This property may not be supported by all editor frameworks.
+	 * </p>
+	 */
+	private Integer escapePosition;
+
+	/**
+	 * Additional text replacements to apply when this proposal is selected.
+	 * <p>
+	 * This property may not be supported by all editor frameworks.
+	 * </p>
+	 */
+	private final List<ReplaceRegion> textReplacements = new ArrayList<ReplaceRegion>();
```

Comment:
in the old class type of field and getters was arraylist.
ok to change?

Reply:
If it's only the field, not public getter, it should be fine

Reply:
yes but there is a noreference on the class